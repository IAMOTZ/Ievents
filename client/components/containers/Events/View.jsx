import React from 'react';
import PropTypes from 'prop-types';
import SideNavigation from '../../common/SideNavigation';
import Header from '../../common/Header';
import EventCards from '../../common/EventCards';
import { AuthTopNavigation } from '../../common/TopNavigation';
import { ConfirmationModal } from '../../common/Modals';
import { LoadingBox, LoadingIcon } from '../../common/LoadingAnimation';

const View = props => (
  <div id="events-container">
    <AuthTopNavigation
      name={props.userName}
      title="My Events"
      isAdmin={props.isAdmin}
      isSuperAdmin={props.isSuperAdmin}
      dispatch={props.dispatch}
    />
    <div className="container-fluid">
      <div className="row">
        <SideNavigation
          name={props.userName}
          isAdmin={props.isAdmin}
          isSuperAdmin={props.isSuperAdmin}
          dispatch={props.dispatch}
        />
        <div className="col-lg-10 offset-md-2 mt-lg-0">
          <Header text="My Events" />
          <div id="page-content">
            <div className="bg-primary rounded text-center mx-auto mt-2 w-lg-50">
              <span className="text-white">Events that their date has passed is considered done</span>
            </div>
            <LoadingIcon start={props.deletingEventStarted} size={2} />
            {
              props.events.length === 0 && props.fetchingEventsStarted ?
                <LoadingBox iconSize={4} /> :
                <div className="mt-5">
                  <div className="card-columns mx-auto">
                    <EventCards
                      events={props.events}
                      centers={props.centers}
                      startDelete={props.startDelete}
                      edit={props.onEdit}
                    />
                  </div>
                </div>
            }
            <ConfirmationModal
              onCancel={props.cancelDelete}
              onOK={props.finishDelete}
            ><span>Are you sure you want to delete this event?</span>
            </ConfirmationModal>
          </div>

        </div>
      </div>
    </div>
    <footer className="d-block d-sm-none mt-5">
      <div className="container text-white text-center py-5">
        <h1>iEvents</h1>
        <p>Copyright &copy; 2017</p>
      </div>
    </footer>
  </div>
);

/* eslint-disable react/forbid-prop-types */
View.propTypes = {
  userName: PropTypes.string.isRequired,
  isAdmin: PropTypes.bool.isRequired,
  isSuperAdmin: PropTypes.bool.isRequired,
  dispatch: PropTypes.func.isRequired,
  deletingEventStarted: PropTypes.bool.isRequired,
  fetchingEventsStarted: PropTypes.bool.isRequired,
  events: PropTypes.array.isRequired,
  centers: PropTypes.array.isRequired,
  startDelete: PropTypes.func.isRequired,
  onEdit: PropTypes.func.isRequired,
  cancelDelete: PropTypes.func.isRequired,
  finishDelete: PropTypes.func.isRequired,
};

export default View;
