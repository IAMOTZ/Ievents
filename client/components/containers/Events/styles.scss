#events-container {
  #page-content {
    min-height: 100vh;
    .fa.fa-trash, .fa.fa-pencil {
      font-size: 20px;
    }
  }
}