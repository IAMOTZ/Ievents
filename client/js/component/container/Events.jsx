import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';

import {
  getAllEvents,
  deleteEvent,
  initializeEdit,
  clearStatus
} from '../../actions/eventActions';
import { getAllCenters } from '../../actions/centerActions';

import UserSideNav from '../common/SideNavigation.jsx';
import { UserTopNav } from '../common/TopNavigation.jsx';
import { ConfirmModal } from '../common/Modal';
import Header from '../common/Header.jsx';
import EventCards from '../common/EventCards.jsx';

@connect((store) => {
  return {
    user: store.user.user,
    authenticated: store.user.status.fetched,
    events: store.events.events,
    centers: store.centers.centers,
    eventDeleted: store.events.status.deleted,
  }
})

export default class Events extends React.Component {
  constructor() {
    super();
    this.state = {
      toDelete: null,
      modalVisible: false,
    }
  }

  // Getting all the events as soon as this component is about to mount the DOM
  componentWillMount() {
    this.props.dispatch(getAllEvents(this.props.user.token));
    this.props.dispatch(getAllCenters());
  }

  // Getting all the events again as soon as this component is updated
  componentDidUpdate() {
    if (this.props.eventDeleted) {
      this.props.dispatch(getAllEvents(this.props.user.token));
      this.props.dispatch(clearStatus('DELETE'));
    }
  }

  // This method simply keep track of an event to be deleted and also 
  // initiate a modal
  startDelete = (e) => {
    const id = e.target.id;
    const state = this.state;
    state.toDelete = id;
    state.modalVisible = !this.state.modalVisible;
    this.setState(state);
  }

  // This method eventually deletes the event and hides back the modal
  finishDelete = (e) => {
    this.props.dispatch(deleteEvent(this.state.toDelete, this.props.user.token));
    this.setState({
      toDelete: null,
      modalVisible: !this.state.modalVisible,
    })
  }

  // This method cancels the deleting and also hides back the modal
  cancelDelete = () => {
    this.setState({
      toDelete: null,
      modalVisible: !this.state.modalVisible,
    });
  }

  // This method simply removes the event without any warning
  removeEvent = (e) => {
    this.props.dispatch(deleteEvent(e.target.id, this.props.user.token));
  }

  // This method initialize editing of an event
  onEdit = (e) => {
    this.props.dispatch(initializeEdit(e.target.id))
  }

  render() {
    if (!this.props.authenticated) {
      return (<Redirect to="/users/login" />)
    } else {
      return (
        <div id="events-container">
          <UserTopNav name={this.props.user.name} title='My Events' />

          <div className="container-fluid">
            <div className="row">

              <UserSideNav userName={this.props.user.name} />

              {/* Main content */}
              <div class="col-lg-10 offset-md-2 mt-lg-0" id="main-content">

                {/* Content Header(navigation) on large screen */}
                <Header text='My Events' />

                {/* Event Grid */}
                <div className="mt-5">
                  <div className="card-columns mx-auto">
                    <EventCards
                      events={this.props.events}
                      centers={this.props.centers}
                      startDelete={this.startDelete}
                      remove={this.removeEvent}
                      edit={this.onEdit} />

                  </div>
                </div>

                <ConfirmModal visible={this.state.modalVisible}
                  onCancel={this.cancelDelete}
                  onOK={this.finishDelete}
                  children="Are you sure you want to delete this event?" />

              </div>
            </div>
          </div>

          <footer class="d-block d-sm-none mt-5">
            <div class="container text-white text-center py-5">
              <h1>Ievents</h1>
              <p>Copyright &copy; 2017</p>
            </div>
          </footer>
        </div>
      )
    }
  }
}