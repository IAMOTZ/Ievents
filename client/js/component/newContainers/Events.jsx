import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import {
  getAllEvents, deleteEvent, setEventToUpdate,
} from '../../actions/eventActions';
import { getAllCenters } from '../../actions/centerActions';
import { stopAsyncProcess } from '../../actions/commonActions';
import * as asyncProcess from '../../actions/asyncProcess';
import UserSideNav from '../common/SideNavigation';
import Header from '../common/Header';
import EventCards from '../common/EventCards';
import { UserTopNav } from '../common/TopNavigation';
import { ConfirmModal } from '../common/Modal';
import { LoadingContainer, LoadingIcon } from '../common/LoadingAnimation';

@connect((store) => {
  const { user } = store.authReducer;
  return {
    user,
    isAdmin: user.role === 'admin',
    isSuperAdmin: user.role === 'admin' || user.role === 'superAdmin',
    isUserAuthenticaed: store.authReducer.loggingUserResolved,
    fetchingEventsStarted: store.fetchEventsReducer.fetchingEventsStarted,
    deletingEventStarted: store.deleteEventReducer.deletingEventStarted,
    deletingEventResolved: store.deleteEventReducer.deletingEventResolved,
    events: store.fetchEventsReducer.events,
    centers: store.fetchCentersReducer.centers,
  };
})
class Events extends React.Component {
  constructor() {
    super();
    this.state = {
      toDelete: null,
      modalVisible: false,
    };
  }

  componentWillMount() {
    this.props.dispatch(getAllEvents(this.props.user.token));
    this.props.dispatch(getAllCenters());
  }

  componentDidUpdate() {
    if (this.props.deletingEventResolved) {
      const userToken = this.props.user.token;
      this.props.dispatch(getAllEvents(userToken));
      this.props.dispatch(stopAsyncProcess(asyncProcess.DELETING_EVENT));
    }
  }

  /**
   * It updates the store about an event that is to be edited.
   * @param {Event} event The event object.
   */
  onEdit = (event) => {
    this.props.dispatch(setEventToUpdate(event.target.id));
  }

  /**
   * It updates the state about an event that is to be deleted.
   * It also initiates the confirmation modal.
   * @param {Event} event The event object.
   */
  startDelete = (event) => {
    const { id } = event.target;
    const state = { ...this.state };
    state.toDelete = id;
    state.modalVisible = !this.state.modalVisible;
    this.setState(state);
  }

  /**
   * It eventually deletes the event and hides back the modal.
   */
  finishDelete = () => {
    this.props.dispatch(deleteEvent(this.state.toDelete, this.props.user.token));
    this.setState({
      toDelete: null,
      modalVisible: !this.state.modalVisible,
    });
    window.scrollTo(0, 0);
  }

  /**
   * This method cancels the deleting and also hides back the modal.
   */
  cancelDelete = () => {
    this.setState({
      toDelete: null,
      modalVisible: !this.state.modalVisible,
    });
  }

  render() {
    if (!this.props.isUserAuthenticaed) {
      return (<Redirect to="/users/login" />);
    }
    return (
      <div id="events-container">
        <UserTopNav
          name={this.props.user.name}
          title="My Events"
          isAdmin={this.props.isAdmin}
          isSuperAdmin={this.props.isSuperAdmin}
          dispatch={this.props.dispatch}
        />
        <div className="container-fluid">
          <div className="row">
            <UserSideNav
              name={this.props.user.name}
              isAdmin={this.props.isAdmin}
              isSuperAdmin={this.props.isSuperAdmin}
              dispatch={this.props.dispatch}
            />
            {/* Main content */}
            <div className="col-lg-10 offset-md-2 mt-lg-0" id="main-content">
              {/* Content Header(navigation) on large screen */}
              <Header text="My Events" />
              <div className="bg-primary rounded text-center mx-auto mt-2 w-lg-50">
                <span className="text-white">Events that their date has passed is considered done</span>
              </div>
              <LoadingIcon start={this.props.deletingEventStarted} size={2} />
              {
                this.props.events.length === 0 && this.props.deletingEventStarted ?
                  <LoadingContainer iconSize={4} /> :
                  <div className="mt-5">
                    <div className="card-columns mx-auto">
                      <EventCards
                        events={this.props.events}
                        centers={this.props.centers}
                        startDelete={this.startDelete}
                        remove={this.removeEvent}
                        edit={this.onEdit}
                      />
                    </div>
                  </div>
              }
              <ConfirmModal
                visible={this.state.modalVisible}
                onCancel={this.cancelDelete}
                onOK={this.finishDelete}
              ><span>Are you sure you want to delete this event?</span>
              </ConfirmModal>
            </div>
          </div>
        </div>
        <footer className="d-block d-sm-none mt-5">
          <div className="container text-white text-center py-5">
            <h1>Ievents</h1>
            <p>Copyright &copy; 2017</p>
          </div>
        </footer>
      </div>
    );
  }
}

export default Events;
