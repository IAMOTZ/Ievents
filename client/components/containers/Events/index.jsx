/* global $ */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  getAllEvents, deleteEvent, setEventToUpdate,
} from '../../../actions/eventActions';
import { stopAsyncProcess } from '../../../actions/commonActions';
import * as asyncProcess from '../../../actions/asyncProcess';
import View from './View';

@connect((store) => {
  const { user } = store.authReducer;
  return {
    userName: user.name,
    userToken: user.token,
    isAdmin: user.role === 'admin' || user.role === 'superAdmin',
    isSuperAdmin: user.role === 'superAdmin',
    fetchingEventsStarted: store.fetchEventsReducer.fetchingEventsStarted,
    deletingEventStarted: store.deleteEventReducer.deletingEventStarted,
    deletingEventResolved: store.deleteEventReducer.deletingEventResolved,
    events: store.fetchEventsReducer.events,
    pagination: store.fetchEventsReducer.pagination
  };
})
class Events extends React.Component {
  constructor() {
    super();
    this.state = {
      toDelete: null,
      modalContent: null,
    };
  }

  componentWillMount() {
    this.props.dispatch(getAllEvents(this.props.userToken, {
      limit: this.props.pagination.limit,
      offset: this.props.pagination.offset,
    }));
  }

  componentDidUpdate() {
    if (this.props.deletingEventResolved) {
      const { userToken } = this.props;
      this.props.dispatch(getAllEvents(userToken, {
        limit: this.props.pagination.limit,
        offset: this.props.pagination.offset,
      }));
      this.props.dispatch(stopAsyncProcess(asyncProcess.DELETING_EVENT));
    }
  }

  /**
   * It updates the store about an event that is to be edited.
   * @param {Event} event The event object.
   */
  onEdit = (event) => {
    const { id } = event.currentTarget;
    this.props.dispatch(setEventToUpdate(id));
  }

  /**
   * It updates the state about an event that is to be deleted.
   * It also initiates the confirmation modal.
   * @param {Event} event The event object.
   */
  startDelete = (event) => {
    const { id } = event.currentTarget;
    const state = { ...this.state };
    state.toDelete = id;
    this.setState(state);
  }

  /**
   * It eventually deletes the event and hides back the modal.
   */
  finishDelete = () => {
    this.props.dispatch(deleteEvent(this.state.toDelete, this.props.userToken));
    this.setState({ toDelete: null });
    $('#confirmation-modal').modal('hide');
    window.scrollTo(0, 0);
  }

  /**
   * This method cancels the deleting and also hides back the modal.
   */
  cancelDelete = () => {
    this.setState({ toDelete: null });
  }

  /**
   * Compose the content to be displayed in the event details modal.
   * @param {Object} event The event whose details is to be displayed.
   */
  createModalContent = (event) => {
    const { title, description } = event;
    const state = { ...this.state };
    state.modalContent = {
      eventTitle: title,
      eventDescription: description,
    };
    this.setState(state);
  }

  /**
   * Updates the pagination for the events.
   * @param {Object} pageData The current page data.
   */
  updatePagination = (pageData) => {
    const nextOffset = pageData.selected * this.props.pagination.limit;
    this.props.dispatch(getAllEvents(this.props.userToken, {
      limit: this.props.pagination.limit,
      offset: nextOffset,
    }));
  }

  render() {
    return (
      <View
        userName={this.props.userName}
        isAdmin={this.props.isAdmin}
        isSuperAdmin={this.props.isSuperAdmin}
        dispatch={this.props.dispatch}
        deletingEventStarted={this.props.deletingEventStarted}
        fetchingEventsStarted={this.props.fetchingEventsStarted}
        events={this.props.events}
        startDelete={this.startDelete}
        onEdit={this.onEdit}
        cancelDelete={this.cancelDelete}
        finishDelete={this.finishDelete}
        pagination={this.props.pagination}
        updatePagination={this.updatePagination}
        modalContent={this.state.modalContent}
        createModalContent={this.createModalContent}
      />
    );
  }
}

Events.defaultProps = {
  userName: '',
  userToken: '',
  isAdmin: false,
  isSuperAdmin: false,
  fetchingEventsStarted: false,
  deletingEventStarted: false,
  deletingEventResolved: false,
  events: [],
  dispatch: () => { },
  pagination: {}
};

/* eslint-disable react/forbid-prop-types */
Events.propTypes = {
  userName: PropTypes.string,
  userToken: PropTypes.string,
  isAdmin: PropTypes.bool,
  isSuperAdmin: PropTypes.bool,
  fetchingEventsStarted: PropTypes.bool,
  deletingEventStarted: PropTypes.bool,
  deletingEventResolved: PropTypes.bool,
  events: PropTypes.array,
  dispatch: PropTypes.func,
  pagination: PropTypes.object,
};

export default Events;
