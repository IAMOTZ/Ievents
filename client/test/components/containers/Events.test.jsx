import React from 'react';
import Events from '../../../components/containers/Events';
import EventsView from '../../../components/containers/Events/View';

const sampleEvent = {
  id: 1,
  centerId: 1,
  userId: 1,
  title: 'test-event',
  description: 'test description',
  date: '12/34/2020',
  status: 'allowed',
};

describe('<Events />', () => {
  const props = {
    userName: 'tester',
    userToken: 'sampleToken',
    isAdmin: false,
    isSuperAdmin: false,
    fetchingEventsStarted: false,
    deletingEventStarted: false,
    deletingEventResolved: false,
    events: [sampleEvent],
    pagination: {},
    dispatch: jest.fn(),
  };
  const alterProps = newProps => ({ ...props, ...newProps });
  describe('Rendering:', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Events {...props} />);
    });
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Behaviour:', () => {
    let wrapper; let wrapperInstance;
    beforeEach(() => {
      props.dispatch.mockReset();
      wrapper = shallow(<Events {...props} />);
      wrapperInstance = wrapper.instance();
    });
    // The names on the following test block refers
    // to the instance method that handles the behaviour tested.
    describe('COMOPONENT_WILL_MOUNT', () => {
      it('should dispatch an action', () => {
        // As soon as the component is shallow rendered
        // the componentWillMount lifecycle method is called.
        expect(props.dispatch.mock.calls.length).toEqual(1);
      });
    });
    describe('COMOPONENT_DID_UPDATE', () => {
      it('should dispatch two action if deleting event resolved', () => {
        props.dispatch.mockReset();
        wrapper.setProps(alterProps({ deletingEventResolved: true }));
        expect(props.dispatch.mock.calls.length).toEqual(2);
      });
    });
    describe('ON_EDIT', () => {
      it('should dispatch an action', () => {
        props.dispatch.mockReset();
        const fakeEvent = { currentTarget: { id: sampleEvent.id } };
        wrapperInstance.onEdit(fakeEvent);
        expect(props.dispatch.mock.calls.length).toEqual(1);
      });
    });
    describe('START_DELETE', () => {
      it('should update the state with id of event to delete', () => {
        const fakeEvent = { currentTarget: { id: sampleEvent.id } };
        wrapperInstance.startDelete(fakeEvent);
        expect(wrapper.state('toDelete')).toEqual(sampleEvent.id);
      });
    });

    describe('FINISH_DELETE', () => {
      it('should dispatch an action and remove the event to delete form state', () => {
        props.dispatch.mockReset();
        wrapperInstance.finishDelete();
        expect(props.dispatch.mock.calls.length).toEqual(1);
        expect(wrapper.state('toDelete')).toBeNull();
      });
    });

    describe('CANCEL_DELETE', () => {
      it('should remove the id of event to delete form state', () => {
        wrapperInstance.cancelDelete();
        expect(wrapper.state('toDelete')).toBeNull();
      });
    });

    describe('CREATE_MODAL_CONTENT', () => {
      it('should update the state with the content of a modal', () => {
        wrapperInstance.createModalContent(sampleEvent);
        expect(wrapper.state('modalContent')).toEqual({
          eventTitle: sampleEvent.title,
          eventDescription: sampleEvent.description,
        });
      });
    });

    describe('UPDATE_PAGINATION', () => {
      it('should dispatch an action', () => {
        props.dispatch.mockReset();
        const fakePageData = { selected: 3 };
        wrapperInstance.updatePagination(fakePageData);
        expect(props.dispatch.mock.calls.length).toEqual(1);
      });
    });
  });
});

describe('<EventsView />', () => {
  const props = {
    userName: 'tester',
    isAdmin: false,
    isSuperAdmin: false,
    dispatch: jest.fn(),
    deletingEventStarted: false,
    fetchingEventsStarted: false,
    events: [],
    startDelete: jest.fn(),
    onEdit: jest.fn(),
    cancelDelete: jest.fn(),
    finishDelete: jest.fn(),
    pagination: {},
    updatePagination: jest.fn(),
    modalContent: {},
    createModalContent: jest.fn(),
  };
  const alterProps = newProps => ({ ...props, ...newProps });
  describe('Rendering', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<EventsView {...props} />);
    });
    it('should render correctly when there are no events and it is fetching', () => {
      wrapper.setProps(alterProps({ fetchingEventsStarted: true }));
      expect(wrapper).toMatchSnapshot();
    });
    it('should render correctly when there are events and it is not fetching', () => {
      wrapper.setProps(alterProps({ events: [sampleEvent] }));
      expect(wrapper).toMatchSnapshot();
    });
  });
});
