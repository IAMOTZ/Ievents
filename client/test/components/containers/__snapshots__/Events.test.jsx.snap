// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Events /> Rendering: should render correctly 1`] = `
<View
  cancelDelete={[Function]}
  createModalContent={[Function]}
  deletingEventStarted={false}
  dispatch={
    [MockFunction] {
      "calls": Array [
        Array [
          [Function],
        ],
      ],
    }
  }
  events={
    Array [
      Object {
        "centerId": 1,
        "date": "12/34/2020",
        "description": "test description",
        "id": 1,
        "status": "allowed",
        "title": "test-event",
        "userId": 1,
      },
    ]
  }
  fetchingEventsStarted={false}
  finishDelete={[Function]}
  isAdmin={false}
  isSuperAdmin={false}
  modalContent={null}
  onEdit={[Function]}
  pagination={Object {}}
  startDelete={[Function]}
  updatePagination={[Function]}
  userName="tester"
/>
`;

exports[`<EventsView /> Rendering should render correctly when there are events and it is not fetching 1`] = `
<div
  id="events-container"
>
  <AuthTopNavigation
    dispatch={[MockFunction]}
    isAdmin={false}
    isSuperAdmin={false}
    name="tester"
    title="My Events"
  />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <SideNavigation
        dispatch={[MockFunction]}
        isAdmin={false}
        isSuperAdmin={false}
        name="tester"
      />
      <div
        className="col-lg-10 offset-md-2 mt-lg-0"
      >
        <Header
          text="My Events"
        />
        <div
          className="page-content"
        >
          <LoadingIcon
            size={2}
            start={false}
          />
          <div
            className="row"
          >
            <EventCards
              createModalContent={[MockFunction]}
              edit={[MockFunction]}
              events={
                Array [
                  Object {
                    "centerId": 1,
                    "date": "12/34/2020",
                    "description": "test description",
                    "id": 1,
                    "status": "allowed",
                    "title": "test-event",
                    "userId": 1,
                  },
                ]
              }
              startDelete={[MockFunction]}
            />
          </div>
          <Pagination
            onPageChange={[MockFunction]}
            pageCount={NaN}
          />
          <ConfirmationModal
            onCancel={[MockFunction]}
            onOK={[MockFunction]}
          >
            <span>
              Are you sure you want to delete this event?
            </span>
          </ConfirmationModal>
          <EventDetailsModal
            eventDescription={null}
            eventTitle={null}
          />
        </div>
      </div>
    </div>
  </div>
  <span
    className="d-block d-sm-none mt-5"
  >
    <Footer />
  </span>
</div>
`;

exports[`<EventsView /> Rendering should render correctly when there are no events and it is fetching 1`] = `
<div
  id="events-container"
>
  <AuthTopNavigation
    dispatch={[MockFunction]}
    isAdmin={false}
    isSuperAdmin={false}
    name="tester"
    title="My Events"
  />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <SideNavigation
        dispatch={[MockFunction]}
        isAdmin={false}
        isSuperAdmin={false}
        name="tester"
      />
      <div
        className="col-lg-10 offset-md-2 mt-lg-0"
      >
        <Header
          text="My Events"
        />
        <div
          className="page-content"
        >
          <LoadingIcon
            size={2}
            start={false}
          />
          <LoadingBox
            iconSize={4}
          />
          <Pagination
            onPageChange={[MockFunction]}
            pageCount={NaN}
          />
          <ConfirmationModal
            onCancel={[MockFunction]}
            onOK={[MockFunction]}
          >
            <span>
              Are you sure you want to delete this event?
            </span>
          </ConfirmationModal>
          <EventDetailsModal
            eventDescription={null}
            eventTitle={null}
          />
        </div>
      </div>
    </div>
  </div>
  <span
    className="d-block d-sm-none mt-5"
  >
    <Footer />
  </span>
</div>
`;
